protocol/x/clob/e2e/isolated_subaccount_orders_test.go

```diff
@@ -0,0 +1,494 @@
+				constants.ClobPair_Btc,
+				constants.ClobPair_Eth,
+				constants.ClobPair_3_Iso,
+			},
+			// Orders should match.
+			orders: []clobtypes.MsgPlaceOrder{
+				PlaceOrder_Alice_Num0_Id0_Clob3_Buy_1ISO_Price10_GTB5,
+				PlaceOrder_Bob_Num0_Id0_Clob3_Sell_1ISO_Price10_GTB5,
+			},
+			collateralPoolBalances: map[string]int64{
+				satypes.ModuleAddress.String(): 10_000_000_000, // $10,000 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 30_000_000_000, // $30,000 USDC
+			},
+			expectedOrdersFilled: []clobtypes.OrderId{},
+			expectedSubaccounts: []satypes.Subaccount{
+				Alice_Num0_MoreIsolatedAfterMatch,
+				Bob_Num0_CrossAfterMatch,
+			},
+			// No changes as no match should have happened.
+			expectedCollateralPoolBalances: map[string]int64{
+				// $10,000 USDC + $10,000 USDC + (match) 100 quote quantums - fee (1 quote quantum)
+				satypes.ModuleAddress.String(): 20_000_000_099,
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+					// $30,000 USDC - $10,000 USDC - (match) 100 quote quantums
+				).String(): 19_999_999_900,
+			},
+		},
+		`Empty subaccount will not have matches for isolated market orders if cross collateral pool does not 
+		have enough balance`: {
+			subaccounts: []satypes.Subaccount{
+				constants.Alice_Num0_10_000USD,
+				constants.Bob_Num0_10_000USD,
+			},
+			perpetuals: []perptypes.Perpetual{
+				constants.BtcUsd_20PercentInitial_10PercentMaintenance,
+				constants.EthUsd_20PercentInitial_10PercentMaintenance,
+				constants.IsoUsd_IsolatedMarket,
+			},
+			clobPairs: []clobtypes.ClobPair{
+				constants.ClobPair_Btc,
+				constants.ClobPair_Eth,
+				constants.ClobPair_3_Iso,
+			},
+			orders: []clobtypes.MsgPlaceOrder{
+				PlaceOrder_Alice_Num0_Id0_Clob3_Buy_1ISO_Price10_GTB5,
+				PlaceOrder_Bob_Num0_Id0_Clob3_Sell_1ISO_Price10_GTB5,
+			},
+			collateralPoolBalances: map[string]int64{
+				satypes.ModuleAddress.String(): 1_000_000_000, // $1,000 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 30_000_000_000, // $30,000 USDC
+			},
+			expectedOrdersFilled: []clobtypes.OrderId{},
+			expectedSubaccounts: []satypes.Subaccount{
+				constants.Alice_Num0_10_000USD,
+				constants.Bob_Num0_10_000USD,
+			},
+			// No changes
+			expectedCollateralPoolBalances: map[string]int64{
+				satypes.ModuleAddress.String(): 1_000_000_000, // $1,000 USDC
+				authtypes.NewModuleAddress(
+					satypes.ModuleName + ":" + lib.UintToString(constants.IsoUsd_IsolatedMarket.Params.Id),
+				).String(): 30_000_000_000, // $30,000 USDC
+			},
+			expectedErr: "insufficient funds",
+		},
+		`Isolated subaccount will not have matches to close isolated perpetual position if isolated collateral pool does not 
+		have enough balance`: {
```

Comment:
same here - when would this happen? when withdrawal gating fails?

Reply:
Yep, this is a problem as well.